### PR TITLE
PLAT-794: Tests msgdelivery (L3) package

### PR DESCRIPTION
**Pull request by dgovorukhin**
- UML diagram
- Minor code refactoring
- TestController reworked

### CODE REVIEW

File: ledger-core/network/nds/msgdelivery/api_shipment.go [hunk 1]
```diff
@@ -6,24 +6,41 @@
...
+}
+
+type ReceiverFunc func(ReturnAddress, nwapi.PayloadCompleteness, interface{}) error
+
+type ShipmentRequest struct {
+	ReceiveFn ReceiverFunc
+	Cancel    *synckit.ChainedCancel
+}
+
+type ShipmentID uint64 // NodeId + ShortShipmentID
```
Comment (cyraxred) [inline ledger-core/network/nds/msgdelivery/api_shipment.go:35]: Please put type _after_ construction methods

File: ledger-core/network/nds/msgdelivery/controller.go [hunk 2]
```diff
@@ -22,8 +22,12 @@ import (
 
 const minHeadBatchWeight = 1 << 20
 
-func NewController(pt uniproto.ProtocolType, factory nwapi.DeserializationFactory,
-	receiveFn ReceiverFunc, resolverFn ResolverFunc, logger uniserver.MiniLogger,
+func NewController(
```
Comment (cyraxred) [inline ledger-core/network/nds/msgdelivery/controller.go:25]: please avoid such code-style changes - it is rare to use this style in the current code base, so it is better to keep it in the original form.

File: ledger-core/network/nds/msgdelivery/ctl_retry_state.go [hunk 3]
```diff
@@ -54,6 +54,8 @@ func (p *stateSender) init(oobQueue, jobQueue int) {
 	p.oob = make(chan rqShipment, oobQueue)
 	p.states = make(chan stateJob, oobQueue)
 	p.jobs = make(chan retryJob, jobQueue)
+	p.requests = make(map[ShipmentID]rqShipment)
```
Comment (cyraxred) [inline ledger-core/network/nds/msgdelivery/ctl_retry_state.go:57]: Why do you do this?

Comment (dgovorukhin) [reply ledger-core/network/nds/msgdelivery/ctl_retry_state.go:57]: (*stateSender).put will not be working without map initialization.

File: ledger-core/network/nds/msgdelivery/controller.go (hunk 2 above)
Comment (dgovorukhin) [reply ledger-core/network/nds/msgdelivery/controller.go]: @cyraxred Fixed

File: ledger-core/network/nds/msgdelivery/api_shipment.go (hunk 1 above)
Comment (dgovorukhin) [reply ledger-core/network/nds/msgdelivery/api_shipment.go]: @cyraxred Fixed

File: ledger-core/network/nds/msgdelivery/ctl_retry_state.go (hunk 3 above)
Comment (cyraxred) [reply ledger-core/network/nds/msgdelivery/ctl_retry_state.go:57]: ok, my bad

File: ledger-core/network/nds/uniproto/l2/uniserver/peer_transport.go [hunk 4]
```diff
@@ -108,6 +112,122 @@ type PeerTransport struct {
 	connections []io.Closer
 }
 
+type rwLockTracker struct {
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l2/uniserver/peer_transport.go:115]: not for working code

**Review by cyraxred — CHANGES_REQUESTED** [review]

File: ledger-core/network/nds/uniproto/l2/uniserver/peer_transport.go (hunk 4 above)
Comment (dgovorukhin) [reply ledger-core/network/nds/uniproto/l2/uniserver/peer_transport.go]: Already removed

**Review by cyraxred — APPROVED** [review]

**Review by AndreyBronin — APPROVED** [review]

**Pull request merged**